waku/factory/app.nim

```diff
@@ -489,8 +317,8 @@ proc setupMonitoringAndExternalInterfaces*(app: var App): AppResult[void] =
 # App shutdown
 
 proc stop*(app: App): Future[void] {.async: (raises: [Exception]).} =
-  if app.restServer.isSome():
-    await app.restServer.get().stop()
+  if app.conf.rest:
+    await app.restServer.stop()
```

Comment:
Don't really see where app.restServer is assigned? 
So when it stops it will result in segfault:

```
/home/nzp/dev/status/nwaku-second/apps/wakunode2/wakunode2.nim(203) wakunode2
/home/nzp/dev/status/nwaku-second/vendor/nim-chronos/chronos/internal/asyncengine.nim(1210) runForever
/home/nzp/dev/status/nwaku-second/vendor/nim-chronos/chronos/internal/asyncengine.nim(1031) poll
/home/nzp/dev/status/nwaku-second/vendor/nim-chronos/chronos/ioselects/ioselectors_epoll.nim(643) selectInto2
../sysdeps/unix/sysv/linux/epoll_wait.c(30) epoll_wait
/home/nzp/dev/status/nwaku-second/apps/wakunode2/wakunode2.nim(175) handleCtrlC
/home/nzp/dev/status/nwaku-second/apps/wakunode2/wakunode2.nim(165) asyncStopper
/home/nzp/dev/status/nwaku-second/vendor/nim-chronos/chronos/internal/asyncfutures.nim(382) futureContinue
/home/nzp/dev/status/nwaku-second/apps/wakunode2/wakunode2.nim(166) asyncStopper
/home/nzp/dev/status/nwaku-second/waku/factory/app.nim(320) stop
/home/nzp/dev/status/nwaku-second/vendor/nim-chronos/chronos/internal/asyncfutures.nim(382) futureContinue
/home/nzp/dev/status/nwaku-second/waku/factory/app.nim(321) stop
/home/nzp/dev/status/nwaku-second/waku/waku_api/rest/server.nim(191) stop
/home/nzp/dev/status/nwaku-second/vendor/nim-chronos/chronos/internal/asyncfutures.nim(382) futureContinue
/home/nzp/dev/status/nwaku-second/waku/waku_api/rest/server.nim(193) stop
/home/nzp/dev/status/nwaku-second/apps/wakunode2/wakunode2.nim(194) handleSigsegv
Segmentation fault
```

Reply:
> Don't really see where app.restServer is assigned? So when it stops it will result in segfault:
> 
> ```
> /home/nzp/dev/status/nwaku-second/apps/wakunode2/wakunode2.nim(203) wakunode2
> /home/nzp/dev/status/nwaku-second/vendor/nim-chronos/chronos/internal/asyncengine.nim(1210) runForever
> /home/nzp/dev/status/nwaku-second/vendor/nim-chronos/chronos/internal/asyncengine.nim(1031) poll
> /home/nzp/dev/status/nwaku-second/vendor/nim-chronos/chronos/ioselects/ioselectors_epoll.nim(643) selectInto2
> ../sysdeps/unix/sysv/linux/epoll_wait.c(30) epoll_wait
> /home/nzp/dev/status/nwaku-second/apps/wakunode2/wakunode2.nim(175) handleCtrlC
> /home/nzp/dev/status/nwaku-second/apps/wakunode2/wakunode2.nim(165) asyncStopper
> /home/nzp/dev/status/nwaku-second/vendor/nim-chronos/chronos/internal/asyncfutures.nim(382) futureContinue
> /home/nzp/dev/status/nwaku-second/apps/wakunode2/wakunode2.nim(166) asyncStopper
> /home/nzp/dev/status/nwaku-second/waku/factory/app.nim(320) stop
> /home/nzp/dev/status/nwaku-second/vendor/nim-chronos/chronos/internal/asyncfutures.nim(382) futureContinue
> /home/nzp/dev/status/nwaku-second/waku/factory/app.nim(321) stop
> /home/nzp/dev/status/nwaku-second/waku/waku_api/rest/server.nim(191) stop
> /home/nzp/dev/status/nwaku-second/vendor/nim-chronos/chronos/internal/asyncfutures.nim(382) futureContinue
> /home/nzp/dev/status/nwaku-second/waku/waku_api/rest/server.nim(193) stop
> /home/nzp/dev/status/nwaku-second/apps/wakunode2/wakunode2.nim(194) handleSigsegv
> Segmentation fault
> ```

Wow! Thanks for it!
I tried to tackle that in https://github.com/waku-org/nwaku/pull/2623/commits/d448dd6a962793dff4169a67e98fd948e136e3e3

qq: how did you force that `stop` ? Very interesting validation :)